anitya/lib/backends/sourceforge_git.py

```diff
@@ -0,0 +1,137 @@
+
+    @classmethod
+    def get_version_url(cls, project):
+        """Method called to retrieve the url used to check for new version
+        of the project provided, project that relies on the backend of this plugin.
+
+        :arg Project project: project (:obj:`anitya.db.models.Project`):
+        Project object whose backend corresponds to the current plugin.
+
+        :Returns: str: url used for version checking
+        """
+
+        namespace = repo = url = ""
+
+        if project.version_url:
+            version_url = project.version_url.replace("+", r"\+")
+            url = (
+                version_url
+                if version_url.startswith(cls.prefix)
+                else f"{cls.prefix}{version_url}{cls.suffix}"
+            )
+
+        else:
+            namespace, repo = cls.get_namespace_repo(project)
+            url = (
+                f"{cls.prefix}{project.name}{cls.suffix}"
+                if not namespace and repo
+                else f"https://sourceforge.net/p/{namespace}/{repo}/ref/master/tags/"
+            )
+
+        return url
+
+    @classmethod
+    def get_namespace_repo(cls, project):
+        """Method called to retrieve namespace and repo from homepage url
+        :arg Project project: a :class:`anitya.db.models.Project` object whose backend
+            corresponds to the current plugin.
+        :return: string namespace and string repo
+        :return type: two strings
+        """
+
+        namespace = repo = url = ""
+
+        if project.homepage.startswith(cls.prefix):
+            url = project.homepage[len(cls.prefix) :]
+
+        if url:
+            namespace, repo = url.split("/")[0:2]
+            repo = "git" if repo == "ci" else repo
+
+        else:
+            namespace = project.name.replace("+", r"\+") if not namespace else namespace
+
+        return namespace, repo
+
+    @classmethod
+    def get_versions(cls, project):
+        """Method called to retrieve all the versions (that can be found)
+        of the projects provided, project that relies on the backend of
+        this plugin.
+
+        :arg Project project: a :class:`anitya.db.models.Project` object whose backend
+            corresponds to the current plugin.
+        :return: a list of all the possible releases found
+        :return type: list
+        :raise AnityaPluginException: a
+            :class:`anitya.lib.exceptions.AnityaPluginException` exception
+            when the versions cannot be retrieved correctly
+
+        """
+        namespace, repo = cls.get_namespace_repo(project)
+        url = project.get_version_url()
+        git_tag_request = requests.get(url)
+
+        if git_tag_request.status_code == 404:
+            raise AnityaPluginException(
+                'Could not call : "%s" of "%s", with error: %s'
+                % (url, project.name, git_tag_request.status_code)
+            )
+
+        soup = BeautifulSoup(git_tag_request.content, "html.parser")
```

Comment:
The approach with BeatifulSoup is really nice. I wonder why it's not used in some other backends.